### PR TITLE
Support v2 iceberg views with gzip

### PR DESCRIPTION
**Pull request by rohanag12**
Similar to #8848, but for views.

### CODE REVIEW

File: catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java [hunk 1]
```diff
@@ -178,7 +178,8 @@ private EntitySnapshotObj.Builder importIcebergView(
       try {
 
         InputStream input = taskRequest.objectIO().readObject(metadataLocation);
-        if (metadataLocation.requiredPath().endsWith(".gz")) {
+        if (metadataLocation.requiredPath().endsWith(".gz")
+            || metadataLocation.requiredPath().endsWith(".gz.metadata.json")) {
```
Comment (dimas-b) [inline catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java:182]: nit: maybe refactor common logic wrt lines 97-98 into a `private` method?

**Review by dimas-b — APPROVED** [review]
Thanks for you contribution, @rohanag12 !

File: catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java (hunk 1 above)
Comment (rohanag12) [reply catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java]: Done. Also ran `./gradlew spotlessApply` to fix style issues.

**Review by dimas-b — APPROVED** [review]

**Review by snazy — APPROVED** [review]
LGTM! Thanks for the contribution, @rohanag12 !

**Pull request merged**